store/tikv/latch/latch.go

```diff
@@ -0,0 +1,186 @@
+// Copyright 2018 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package latch
+
+import (
+	"fmt"
+	"math"
+	"sort"
+	"sync"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spaolacci/murmur3"
+)
+
+// Latch stores a key's waiting transactions
+type Latch struct {
+	waiting      []uint64
+	lastCommitTs uint64
+	lock         sync.Mutex
```

Comment:
you can use `sync.Mutex` as anonymous field. In this way you can directly call `Lock`  and `Unlock` when you want to use either one of them.  please refer http://dennissuratna.com/using-anonymous-field-to-extend-in-go/ to find more details about anonymous filed.